mmv1/third_party/terraform/data_sources/data_source_google_iam_policy.go.erb

```diff
@@ -120,30 +120,53 @@ func dataSourceGoogleIamPolicyRead(d *schema.ResourceData, meta interface{}) err
 	bset := d.Get("binding").(*schema.Set)
 	aset := d.Get("audit_config").(*schema.Set)
 
-	// All binding{} blocks will be converted and stored in an array
-	bindings = make([]*cloudresourcemanager.Binding, bset.Len())
-	policy.Bindings = bindings
-
 	// Convert each config binding into a cloudresourcemanager.Binding
-	for i, v := range bset.List() {
+	// and merge member lists of equivalent binding{} blocks from the config provided by the user
+	bindingMap := map[string]*cloudresourcemanager.Binding{}
+	for _, v := range bset.List() {
 		binding := v.(map[string]interface{})
 		members := convertStringSet(binding["members"].(*schema.Set))
 		condition := expandIamCondition(binding["condition"])
 
-		// Sort members to get simpler diffs as it's what the API does
-		sort.Strings(members)
+		// Map keys are used to identify binding{} blocks that are identical except for the member lists
+		key := binding["role"].(string)
+		if condition != nil {
+			key += fmt.Sprintf("-[%s]-[%s]-[%s]-[%s]", condition.Expression, condition.Title, condition.Description, condition.Location)
+		}
+		key = strconv.Itoa(hashcode(key))
```

Comment:
why is the conversion string -> hashcode() -> string necessary? could this use the human-readable string instead?

Reply:
Yeah I agree human readable would be better here in case someone uses a debugger. I copied how the state ID is set near the end of the function but it's not as necessary in this map!

Reply:
ah, interesting - I wonder why that uses hashcode? Maybe there's a size limit in terraform?